middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -3086,8 +3105,10 @@ let wrap_final_module_block acc env ~program ~prog_return_cont
     let module_block_simple =
       let simple_var = Simple.var module_block_var in
       match find_value_approximation env simple_var with
-      | Value_approximation.Value_symbol s -> Simple.symbol s
-      | _ -> simple_var
+      | Value_symbol s -> Simple.symbol s
+      | Value_int i -> Simple.const_int i
```

Comment:
I'm not sure about this addition. We're creating a simple for the main module block, if it is an integer we have a serious problem.